### DISCUSSION TITLE
Hard to drive from keyboard: very few keyboard shortcuts, strange tab orders

### DISCUSSION THREAD

**Discussion by llimeht** [SasView/sasview#2342] [General]
**Describe the bug**
In trying to write some GUI tests, I was looking for keyboard shortcuts to perform tasks like load data, switch between panels, etc; as a user, I also look for keyboard acceleration rather than doing more mouse-miles.

Keyboard shortcuts are good for accessibility of the interface, for speeding up common operations, and for automating tasks. It's quite unusual to have so few shortcuts defined.

**Expected behavior**
Sasview has quite few keyboard shortcuts defined. 
* Alt+F opens the file menu; I would expect the other menus to have accelerator keys defined
* I would expect there to be keystrokes to switch to different perspectives
* I would expect shortcuts for file operations like loading data or projects, saving projects 
* I would expect key buttons like the "send to" button to have accelerators defined.

Within a dialogue, the tab order should also correspond to the visual layout so that pressing tab or shift+tab takes the user logically forwards pretty backwards though fields - this isn't the case in the Fitting panel however. (Other panels untested)

All parts of the window should be accessible by the keyboard shortcuts - I would expect to be able to flip between parameters in the fit panel using cursor or pgup/pgdn keys for instance. However, some parts of the fit window didn't seem to be accessible be keyboard at all. For example, in the fit panel, I could not find a way to get from the checkbox that defines a parameter to be fitted to the numerical entry fields for the values, limits etc.

For manual refinement of the fit prior to optimisation, I would expect the values of the parameters to be variable by keystroke in a jog-dial fashion.

A systematic review of existing keyboard shortcuts and then planning out a set of new, logical, non-conflicting shortcuts is needed.

**SasView version (please complete the following information):**
 - Version: 5.0.5 and main HEAD

**Operating system (please complete the following information):**
 - OS: Windows 10 & Linux tested

**Comment by rozyczko** [comment]
This is a valid concern. All GUI apps should have well defined keyboard accelerators and tab ordering.
Keyboard accelerators will have to be agreed upon and well documented, especially for tasks which are not obvious. The obvious ones are file/project load/save, dialog/app close, help etc.: basic widget-oriented tasks.
Less obvious would be shortcuts allowing certain functionality like switching perspective, showing tools and option widgets.

I started looking at the easy part - ordering UI elements for consistent tabbing. Looking at some calculators and utilities, this is in complete disarray and needs to be fixed.
